swiftshield-Sources/SourceKitdResponse.swift

```diff
@@ -50,8 +50,11 @@ public class SourceKitdResponse: CustomStringConvertible {
             return SourceKitdUID(uid: value)
         }
 
-        public func getArray(_ key: SourceKitdUID) -> Array {
+        public func getArray(_ key: SourceKitdUID) -> Array? {
             let value = sourcekitd_variant_dictionary_get_value(dict, key.uid)
+            if value.data.0 == 0 || value.data.1 == 0 {
+                return nil
+            }
```

Comment:
It's unclear to me what this means -- could you clarify why the value being set to `0` means the array is nil?
If the intention of this returning nil is to confirm that there aren't attributes in the dictionary, I feel there's probably a better way to achieve this in sourcekit's API

Reply:
The source of [sourcekitd_variant_dictionary_get_value()](https://github.com/apple/swift/blob/746b58e8e1c1ef1ac09d8031875fd2a08b65597c/tools/SourceKit/tools/sourcekitd/lib/API/sourcekitdAPI-Common.cpp#L347), just assign the result from makeNullVariant(), which is defined [here](https://github.com/apple/swift/blob/746b58e8e1c1ef1ac09d8031875fd2a08b65597c/tools/SourceKit/tools/sourcekitd/include/sourcekitd/Internal.h#L180). I cannot find any API to check whether the data is null. 
The definition of `sourcekitd_variant_t` is C code. The way we are checking here is also like a C style. 
```
typedef struct {
  uint64_t data[3];
} sourcekitd_variant_t;
```

Reply:
Got it!